src/utils/mtev_curl.c

```diff
@@ -255,7 +267,11 @@ eventer_curl_handle_socket(CURL *easy, curl_socket_t s, int action, void *userp,
           mtevL(debug_ls, "curl removed %p as %p\n", c->e, tofree);
           c->e = NULL;
         }
-        free(c);
+        bool zero = false;
```

Comment:
should make a function ``curl_context_deref``, otherwise same as previous comment,